packages/documentation/src/components/sidebar/SidebarItems.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Link, StaticQuery, graphql } from 'gatsby';
+
+/**
+ * Sidebar
+ *
+ * @param {props} props
+ */
+
+function ComponentList() {
+  return (
+    <StaticQuery
+      query={graphql`
+        query ComponentListQuery {
+          components: allSitePage(
+            filter: {context: {source: {eq: "component"}}}
+          ) {
+            edges {
+              node {
+                path
+                context {
+                  source
+                  name
+                }
+              }
+            }
+          }
+        }
+      `}
+      render={data => (
+        <SidebarItems items={data.components.edges.map(comp => {
+          return {
+            href: comp.node.path,
+            name: comp.node.context.name,
+          };
+        })}/>
+      )}/>
+  );
+
+}
+
+export default function SidebarItems({ items }) {
+  if (!items.length) {
+    return null;
+  }
+
+  return (
+    <ul className="menu-list">
+      {items.map(item => {
+        if (item.items) {
+          return (
+            <li key={item.name}>
+              <h4>{item.name}</h4>
+              <SidebarItems items={item.items}/>
+            </li>
+          );
+        }
+
+        if (item.query === 'componentList') {
+          return (
+            <li key={item.name}>
+              <h4>{item.name}</h4>
+              <ComponentList/>
```

Comment:
Not sure if this would be better but we could wrap conditionals around the specific differences here: 

```
{item.href && <li key={item.href}><Link to={item.href}>{item.name}</Link></li>}
{!item.href && 
    <li key={item.name}>
      <h4>{item.name}</h4>
        {item.items && <SidebarItems/>}
        {item.query && <ComponentList/>}
     </li>
}
```

Reply:
Could use an `item.href ? :` but I think the not `item.href` lines up more nicely.

Reply:
I think that generally works better and updated it, but it does leave you at the end with a longer conditional for the final one, if it's a sort of catch all, display if none of the other conditions are hit, thing. I still think the approach you suggested is clearer, though.